### ISSUE TITLE
mbstring.func_overload = 2 + xlsx = file corrupted

### ISSUE THREAD

**Issue by yakovkravtsov** [PHPOffice/PhpSpreadsheet#1951]
1) PHP 7.3
2) Bitrix
3) mbstring.func_overload      = "2"
4) mbstring.internal_encoding  = "UTF-8"

---

XLSX file is created corrupted, nothing helps.

Only after I turned off "mbstring.func_overload" - the file was created and opened normally.

Tried saving to both file and php: // output. The result is the same.

What to do?

**Comment by MarkBaker** [comment]
It raises the question of why you are using `mbstring.func_overload` anyway

Note the warning in the PHP documentation
 > __Warning__ This feature has been DEPRECATED as of PHP 7.2.0. Relying on this feature is highly discouraged.

And some of the comments about its use:

> it's kinda evil and gives you tons of headache.

> I took over a project with this "feature" enabled... took me a while to figure out why strings behaved non language standard. So, if you like to make grown men cry - by all means enable this deathtrap.

If you have to use it, then it is possible to set it on a per-directory basis (although this can create its own problems), so you could set it to 0 (disable it) for the PhpSpreadsheet folder

```
<Directory ...>
   ...
   php_admin_value mbstring.func_overload 0
</Directory>
```

**Comment by gserrg** [comment]
@MarkBaker  The author uses bitrix, which does not work without func_overload. Everyone who works with bitrix should be prepared from the outset that 1/3 of the libraries are not compatible with it.

@yakovkravtsov Since it was required, then there are options:
1. Use phpspreadsheet 1.12
2. As mentioned above - change php_admin_value for a specific directory (it is highly undesirable to load the bitrix kernel in it).
3. Run cli-php via a system call (using the bitrix kernel is also not desirable).
4. Dangerous, but as an extreme option: temporarily change the encoding to single-byte.

**Comment by noopple** [comment]
I just researched and found that issue relates to package maenchen/zipstream-php. I replaced string functions listed [here](https://www.php.net/manual/en/mbstring.overload.php) locally and it resolved the issue. So we can close this ticket.

**Comment by gserrg** [comment]
Duplicate of #1658

**Comment by dev-bitrix** [comment]
i have the same issue, because client wont paid for license and core has old version what works with func_overload
@gserrg thnx for options to resolve problem
me helped 1st option, downgrade version to 1.12 and files what generated with phpspreadsheet not contains corrupt errors

**Comment by oleibman** [comment]
Closing. mbstring.func_overload was removed with Php 8.0.0, and we no longer support any earlier releases.
